src/libAtomVM/term.h

```diff
@@ -86,6 +86,20 @@ extern "C" {
 
 #define TERM_FROM_ATOM_INDEX(atom_index) ((atom_index << 6) | 0xB)
 
+#ifndef TYPEDEF_GLOBALCONTEXT
+#define TYPEDEF_GLOBALCONTEXT
+typedef struct GlobalContext GlobalContext;
+#endif
+
+typedef struct PrinterFun PrinterFun;
+
+typedef int (*printer_function_t)(PrinterFun *fun, const char *fmt, ...) PRINTF_FORMAT_ARGS(2, 3);
```

Comment:
Should the return type be unsigned?  Is there any case where it would be negative?

Reply:
I'm just mimicking printf here: "if an output error is encountered, a negative value is returned."